external_deps/build.sh

```diff
@@ -1,9 +1,10 @@
 #!/usr/bin/env bash
 
-# Exit on error
-# Error on undefined variable
-set -e
-set -u
+# Exit on undefined variable and error.
+set -u -e -o pipefail
+
+SCRIPT_DIR="$(cd "$(dirname "${BASH_SOURCE[0]}")" >/dev/null 2>&1 && pwd)"
+cd "${SCRIPT_DIR}"
```

Comment:
Please no. I often run this in different directories so that I can test things without clobbering the ones that I usually use.

Reply:
Oh? Well…

Reply:
I remove the `cd`.

I kept the `SCRIPT_DIR=` at the top and also moved `WORK_DIR=` at the top because I feel more confident this way, expecially with the `WORK_DIR=` which is calling for trouble if done late in execution in a function. I don't want to be bitten by this in the future after possible rewrites.